### PR TITLE
SLCORE-435 Add param context key and filter the tab contents

### PR DESCRIPTION
**Pull request by mithil-dobarkar-sonarsource**
Please review our [contribution guidelines](https://github.com/SonarSource/sonarlint-core/blob/master/contributing.md).

And please ensure your pull request adheres to the following guidelines: 

- [ ] Please explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [ ] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [ ] Provide a unit test for any code you changed
- [ ] If there is a [JIRA](http://jira.sonarsource.com/browse/SLCORE) ticket available, please make your commits and pull request start with the ticket ID (SLCORE-XXXX)

### CODE REVIEW

File: server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/push/parsing/TaintVulnerabilityRaisedEventParser.java [hunk 1]
```diff
@@ -84,6 +85,8 @@ private static class TaintVulnerabilityRaisedEventPayload {
     private LocationPayload mainLocation;
     private List<FlowPayload> flows;
 
+    private String ruleDescriptionContextKey;
```
Comment (henryju) [inline server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/push/parsing/TaintVulnerabilityRaisedEventParser.java:88]: Minor: you can mark this field as `@Nullable`

File: server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/storage/XodusServerIssueStore.java [hunk 2]
```diff
@@ -118,6 +118,7 @@ public class XodusServerIssueStore implements ProjectServerIssueStore {
   private final Path backupFile;
 
   private final Path xodusDbDir;
+  private static final String RULE_DESCRIPTION_CONTEXT_KEY = "ruleDescriptionContextKey";
```
Comment (henryju) [inline server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/storage/XodusServerIssueStore.java:121]: I would move the constant with other constants, and rename to `RULE_DESCRIPTION_CONTEXT_KEY_PROPERTY_NAME` for consistency

**Review by henryju — CHANGES_REQUESTED** [review]
Please move the implementation code to the `core` module, but overall LGTM!

File: core/src/main/java/org/sonarsource/sonarlint/core/rules/OthersSectionHtmlContent.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+public class OthersSectionHtmlContent {
+
+  private static final String FOLDER_NAME = "/context-rule-description/";
+  private static final String FILE_EXTENSION = ".html";
+  private static final String UNSUPPORTED_RULE_DESCRIPTION_FOR_CONTEXT_KEY = "Unsupported rule description for context key: ";
+  private static final String ERROR_READING_FILE_CONTENT = "Could not read the content for rule description for context key: ";
+  private OthersSectionHtmlContent() {}
+
+  public static String getHtmlContent(String contextKey) {
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/rules/OthersSectionHtmlContent.java:36]: If I'm correct, we only have one possible value for this method parameter? Why not simply changing to:
```suggestion
  public static String getHtmlContent() {
```

**Review by henryju — COMMENTED** [review]
Minor feedback, then LGTM

**Review by henryju — APPROVED** [review]

**Comment by mithil-dobarkar-sonarsource** [comment]
@damien-urruty-sonarsource, Julien has already reviewed my PR, kindly review that I have not removed your changes while resolving the conflicts.

**Pull request merged**